diff --git a/vcg/complex/algorithms/closest.h b/vcg/complex/algorithms/closest.h
old mode 100644
new mode 100755
index 062404d0..c233e0b3
--- a/vcg/complex/algorithms/closest.h
+++ b/vcg/complex/algorithms/closest.h
@@ -204,7 +204,8 @@ namespace vcg {
 
         template <class MESH, class GRID>
             typename MESH::VertexType * GetClosestVertex( MESH & mesh,GRID & gr,const typename GRID::CoordType & _p,
-            const typename GRID::ScalarType & _maxDist,typename GRID::ScalarType & _minDist )
+            const typename GRID::ScalarType & _maxDist,typename GRID::ScalarType & _minDist,
+            typename GRID::CoordType &_closestPt)
         {
             typedef typename GRID::ScalarType ScalarType;
             typedef Point3<ScalarType> Point3x;
@@ -214,7 +215,7 @@ namespace vcg {
             typedef vcg::vertex::PointDistanceFunctor<typename MESH::ScalarType> VDistFunct;
             VDistFunct fn;
             _minDist=_maxDist;
-            Point3x _closestPt;
+//            Point3x _closestPt;
             return (gr.template GetClosest<VDistFunct,MarkerVert>(fn,mv,_p,_maxDist,_minDist,_closestPt));
         }
 
